x-pack/plugins/ml/public/embeddables/anomaly_swimlane/explorer_swimlane_container.tsx

```diff
@@ -0,0 +1,168 @@
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, useEffect, useMemo, useState } from 'react';
+import { EuiResizeObserver, EuiSpacer } from '@elastic/eui';
+import { combineLatest, from, Observable, of, Subject } from 'rxjs';
+import { catchError, debounceTime, map, skipWhile, startWith, switchMap } from 'rxjs/operators';
+import { throttle } from 'lodash';
+import { i18n } from '@kbn/i18n';
+import { CoreStart } from 'kibana/public';
+import { ExplorerSwimlane } from '../../application/explorer/explorer_swimlane';
+import { TimeBuckets } from '../../application/util/time_buckets';
+import { ExplorerJob, OverallSwimlaneData } from '../../application/explorer/explorer_utils';
+import { MlStartDependencies } from '../../plugin';
+import {
+  AnomalySwimlaneEmbeddableInput,
+  AnomalySwimlaneEmbeddableOutput,
+  MlServices,
+} from './anomaly_swimlane_embaddable';
+import { parseInterval } from '../../../common/util/parse_interval';
+import { SWIMLANE_TYPE } from '../../application/explorer/explorer_constants';
+import { ChartTooltip } from '../../application/components/chart_tooltip';
+
+const RESIZE_THROTTLE_TIME_MS = 200;
+
+export interface ExplorerSwimlaneContainerProps {
+  id: string;
+  embeddableInput: Observable<AnomalySwimlaneEmbeddableInput>;
+  services: [CoreStart, MlStartDependencies, MlServices];
+  refresh: Observable<any>;
+  onOutputChange: (output: Partial<AnomalySwimlaneEmbeddableOutput>) => void;
+}
+
+export const ExplorerSwimlaneContainer: FC<ExplorerSwimlaneContainerProps> = ({
+  id,
+  embeddableInput,
+  services,
+  refresh,
+  onOutputChange,
+}) => {
+  const [{ uiSettings, notifications }, , { explorerService }] = services;
+
+  const [swimlaneData, setSwimlaneData] = useState<OverallSwimlaneData>();
+  const [chartWidth, setChartWidth] = useState<number>(0);
+  const [swimlaneType, setSwimlaneType] = useState<string>();
+
+  const chartWidth$ = useMemo(() => new Subject<number>(), []);
+
+  const timeBuckets = new TimeBuckets({
+    'histogram:maxBars': uiSettings.get('histogram:maxBars'),
+    'histogram:barTarget': uiSettings.get('histogram:barTarget'),
+    dateFormat: uiSettings.get('dateFormat'),
+    'dateFormat:scaled': uiSettings.get('dateFormat:scaled'),
+  });
+
+  useEffect(() => {
+    chartWidth$.next(chartWidth);
+  }, [chartWidth]);
+
+  useEffect(() => {
+    const subscription = combineLatest([
+      embeddableInput,
+      chartWidth$.pipe(skipWhile(v => !v)),
+      refresh.pipe(startWith(null)),
+    ])
+      .pipe(
+        debounceTime(500),
+        switchMap(([input, swimlaneContainerWidth]) => {
+          const { jobs, viewBy, swimlaneType: swimlaneTypeInput, timeRange } = input;
+
+          explorerService.setTimeRange(timeRange);
+
+          onOutputChange({ timeRange });
```

Comment:
This is an interesting use case.  Is the idea that the swimlane doesn't want to change the global dashboard time range, as in "only show data from this time range", but instead have it be more like "highlight data in this time range"?

If so I think maybe this still should be input, and we could set it globally, so other ML visualizations can render themselves according to both `timeRange` and `highlightTimeRange` parameters.  It looks like that's how it works in the ML app?

Reply:
It's possible to set a custom time range per panel, but I also wanted to make it persistent, so when the user came back to the dashboard it remembers the selected time range. I managed to achieve it by update the output with the latest provided time range. Could be I used it in a wrong, so suggestions are welcome

Reply:
hmm, this doesn't sound right.  The custom time range per panel _should_ be persistent, without the embeddable needing to do anything special.  Did it not work that way without this extra code?

Reply:
It's possible to customize a time range in "edit" mode and in "view" mode. When the user perform changes in "edit" mode and save the dashboard, the state persists. But it's not the case in the "view" mode. Probably it's expected behavior, I removed this code for the time range output

Reply:
👍  Yep, that is expected, removing this special handling I think is the right call so it works the same across all embeddable types.

Reply:
Fixed in the latest changes